packages/medusa/src/loaders/api.ts

```diff
@@ -1,9 +1,27 @@
-import { Express } from 'express'
+import { Express, Router } from 'express'
 import bodyParser from "body-parser"
 import routes from "../api"
 import { AwilixContainer } from "awilix"
 import { ConfigModule } from "../types/global"
 
+import { awaitMiddleware, checkCustomErrors } from "../api/middlewares";
+import { RequestHandler } from "express-serve-static-core";
+import { PartialRequestHandler } from '../api/middlewares/await-middleware'
+
+["get", "post", "put", "delete"].forEach(function (method: string): void {
```

Comment:
Suggestion: move that to the middleware directory under apply-global-middleware file. Wdyt?